addons/web/static/src/js/core/owl_popover.js

```diff
@@ -0,0 +1,272 @@
+odoo.define('web.OwlPopover', function() {
+    'use strict';
+
+    const { Component, hooks, misc, QWeb } = owl;
+    const { Portal } = misc;
+    const { useExternalListener, useRef, useState } = hooks;
+
+    /**
+     * Popover (owl version)
+     *
+     * Represents a bootstrap-style popover handled with pure JS. The popover
+     * will be visually bound to its `target` using an arrow-like '::before'
+     * CSS pseudo-element.
+     * @extends Component
+     **/
+    class Popover extends Component {
+        /**
+         * @param {Object} [props]
+         * @param {string} [props.position='bottom'] 'top', 'bottom', 'left' or 'right'
+         * @param {HTMLElement} [props.target='body'] the target element (the popover's
+         *                      arrow decoration will point to it)
+         * @param {string} [props.title]
+         */
+        constructor() {
+            super(...arguments);
+            this.popoverRef = useRef('popover');
+            this.orderedPositions = ['top', 'right', 'bottom', 'left'];
+            this.requestedPosition = null;
+            this.state = useState({
+                displayed: false,
+            })
+            // without event capture, scroll listener on children nodes will prevent
+            // this handler from being fired
+            useExternalListener(document, 'scroll', this.show, {
+                capture: true,
+            });
+            useExternalListener(window, 'resize', this.show);
+        }
+
+        patched() {
+            super.patched(...arguments);
+            this.show();
+        }
+
+        /**
+         * Show the popover according to its props. This method will try to position the
+         * popover as request (according to the `position` props) but will fallback
+         * on 'bottom' positioning if the requested position would put it outside of the
+         * viewport.
+         */
+        show() {
+            if (!this.state.displayed) {
+                return;
+            }
+            // get the target from the dom
+            // we don't want to do this early, since the target might not
```

Comment:
early -> earlier?

Reply:
my sentence is grammatically correct; "i don't want to wake up early" is fine